src/queryBuilder/QueryBuilder.js

```diff
@@ -1,67 +1,53 @@
 import React from 'react'
-import { observable } from 'mobx'
+import F from 'futil-js'
 import DDContext from './DragDrop/DDContext'
-import { Component } from '../utils/mobx-react-utils'
 import { Modal as DefaultModal, NestedPicker } from '../layout/'
 import Group from './Group'
 import styles from '../styles'
+import { contexturify } from '../utils/hoc'
+import { useLens } from '../utils/futil'
 
 let { background } = styles
 
 export default DDContext(
-  Component(
-    (
-      { tree: iTree, types: iTypes, typeComponents: iTypeComponents },
-      {
-        typeComponents = iTypeComponents,
-        types = iTypes || typeComponents,
-        tree = iTree,
-      }
-    ) => ({
-      types,
-      state: observable({
-        adding: false,
-        ...tree,
-      }),
-    }),
+  contexturify(
     ({
-      state,
-      path,
+      tree,
+      node,
       fields,
       types = {},
```

Comment:
let's just entirely remove support for this and rely on mapNodeToProps

Reply:
this turned out to be a pretty sizable change, since a lot of stories were using `types`/`typeComponents` and needed to be updated, so I opened a new PR for it instead: #286